### PR TITLE
Mobile client: static

### PR DESCRIPTION
**Pull request by Arthi-chaud**

### CODE REVIEW

File: mobile/lib/I10n/app_en.arb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "helloWorld": "Hello World!",
+    "@helloWorld": {
+      "description": "The conventional newborn programmer greeting"
+    }
+}
```
Comment (Octopus773) [inline mobile/lib/I10n/app_en.arb:6]: what are these files ending in .arb

File: mobile/lib/src/main.dart [hunk 2]
```diff
@@ -1,115 +1,84 @@
...
+            '/pipeline/new': () => const CreatePipelinePage()
+          };
+          Map routes = {}
+            ..addAll(cardRoutes)
+            ..addAll(pageRoutes);
+          return PageRouteBuilder(
+              opaque: false,
+              settings: settings,
+              pageBuilder: (_, __, ___) => routes[settings.name].call(),
+              transitionDuration: const Duration(milliseconds: 500),
```
Comment (Octopus773) [inline mobile/lib/src/main.dart:66]: I don't know what transitions look like are, but 500ms seems a lot for animations

File: mobile/lib/src/models/pipeline_collection.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      enabled.addAll(disabled);
+      disabled.clear();
+    }
+    switch (sortingMethod) {
+      case PipelineCollectionSort.last:
+        sortingFunction = (a, b) {
+          if (b.trigger.last == null) return -1;
+          if (a.trigger.last == null) return 1;
+          return b.trigger.last!.compareTo(a.trigger.last!);
+        };
```
Comment (Octopus773) [inline mobile/lib/src/models/pipeline_collection.dart:45]: please be consistent with your indent

**Review by Octopus773 — COMMENTED** [review]
Very hard pr to look at,
i can only do linter stuff (your indent policy is on vacation, but not that important)
i will test the app on my phone to see if it's great

**Comment by Octopus773** [comment]
I tried to build the dockerfile, and it fails to build @Arthi-chaud 

```
Creating network "aeris_default" with the default driver
Building mobile
Sending build context to Docker daemon  240.6kB
Step 1/6 : FROM mingc/android-build-box:latest
latest: Pulling from mingc/android-build-box
ea362f368469: Pull complete 
ddca2bd6700a: Pull complete 
ebe1dff30b87: Pull complete 
e7d8b6e2fedc: Pull complete 
2b914c2f8186: Pull complete 
2c6da7d4a9a6: Pull complete 
3fdf9fc1f5e5: Pull complete 
e23528d82d22: Pull complete 
3f87dd40b2a0: Pull complete 
7940ca8732f2: Pull complete 
a54899215dc5: Pull complete 
f372187f724a: Pull complete 
93af00d1f580: Pull complete 
ce590e8587a4: Pull complete 
0a0890f8c35f: Pull complete 
d09284b1ffaa: Pull complete 
e15bc121f383: Pull complete 
9c6484ef75cc: Pull complete 
cc3d00200eb6: Pull complete 
48c8207cfbf2: Pull complete 
Digest: sha256:0233f9435394c4cbfb487a7ff5fb3657c5f8f772b71766e175890597774df99c
Status: Downloaded newer image for mingc/android-build-box:latest
 ---> 361ea8aa7389
Step 2/6 : COPY pubspec.* ./
 ---> 8a5136454540
Step 3/6 : RUN flutter pub get
 ---> Running in 4a5740cf4edc
Running "flutter pub get" in flutter_tools...                       7.1s
Running "flutter pub get" in project...                            19.7s
Removing intermediate container 4a5740cf4edc
 ---> 2b1d2a2e0e9b
Step 4/6 : COPY . .
 ---> 4737cb2e8df6
Step 5/6 : RUN flutter build apk lib/src/main.dart --no-pub
 ---> Running in f9b4eea2e41b

💪 Building with sound null safety 💪

Running Gradle task 'assembleRelease'...                        

lib/src/constants.dart:7:44: Error: Required named parameter 'primaryVariant' must be provided.
const ColorScheme aerisScheme = ColorScheme(
                                           ^
/opt/flutter/packages/flutter/lib/src/material/color_scheme.dart:20:9: Context: Found this candidate, but the arguments don't match.
  const ColorScheme({
        ^^^^^^^^^^^
lib/src/views/create_pipeline_page.dart:67:38: Error: The getter 'secondaryContainer' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/opt/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'secondaryContainer'.
                                    .secondaryContainer,
                                     ^^^^^^^^^^^^^^^^^^
lib/src/views/create_pipeline_page.dart:102:34: Error: The getter 'secondaryContainer' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/opt/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'secondaryContainer'.
                                .secondaryContainer,
                                 ^^^^^^^^^^^^^^^^^^
lib/src/views/pipeline_detail_page.dart:98:50: Error: The getter 'secondaryContainer' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/opt/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'secondaryContainer'.
            color: Theme.of(context).colorScheme.secondaryContainer,
                                                 ^^^^^^^^^^^^^^^^^^
lib/src/widgets/warning_dialog.dart:33:60: Error: The getter 'primaryContainer' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/opt/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'primaryContainer'.
                    primary: Theme.of(context).colorScheme.primaryContainer),
                                                           ^^^^^^^^^^^^^^^^
lib/src/widgets/home_page_sort_menu.dart:25:22: Error: Member not found: 'webhook'.
        return Icons.webhook;
                     ^^^^^^^


FAILURE: Build failed with an exception.

* Where:
Script '/opt/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 1070

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/opt/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 3s
Running Gradle task 'assembleRelease'...                          123.8s
Gradle task assembleRelease failed with exit code 1
The command '/bin/sh -c flutter build apk lib/src/main.dart --no-pub' returned a non-zero code: 1
ERROR: Service 'mobile' failed to build : Build failed
```

**Review by 0Nom4D — CHANGES_REQUESTED** [review]
Problems seen have already been reported by @Octopus773.

File: mobile/lib/src/models/pipeline_collection.dart (hunk 3 above)
Comment (0Nom4D) [reply mobile/lib/src/models/pipeline_collection.dart]: ## Fix applied

Identation has been fixed to be compliant with Android Studio identation (2 spaces max 4).

File: mobile/lib/src/main.dart (hunk 2 above)
Comment (0Nom4D) [reply mobile/lib/src/main.dart]: ## Test method

I'm testing values between 225ms and 800ms to check which one is the best looking. I'll let you know.

Comment (0Nom4D) [reply mobile/lib/src/main.dart]: Animation duration value has been modified to 350ms.

**Review by 0Nom4D — APPROVED** [review]
Except for .arb files and animation duration (to be reviewed), no more things have to be changed before any more review.

**Comment by Octopus773** [comment]
The .apk build is now working fine !

**Comment by Octopus773** [comment]
The .apk build is now working fine !

**Review by Octopus773 — APPROVED** [review]

File: mobile/I10n.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+arb-dir: lib/l10n
+template-arb-file: app_en.arb
+output-localization-file: app_localizations.dartà
```
Comment (GitBluub) [inline mobile/I10n.yaml:3]: "à" at the end of the file

**Review by GitBluub — APPROVED** [review]

**Comment by AnonymusRaccoon** [comment]
Honestly, I kind of gave up to review the code due to the size of the PR, but I tested the app, and it's really good. I have a few points where the behavior is kinda off, but this is only nitpicking:
 - The app name is "mobile"
 - The logo is the default flutter one, not the one on the README
 - The pipelines automatically reload when you reach the top, I would expect to have to pull a refresh logo from the top and not simply reach the top to reload
 - Refreshing the view blurs it, and we can't use the app while it reloads. Yes, this is pretty, but this is frustrating for the user.
 - The edit action panel can be disabled if you scroll down from the top but not from the middle. Instead, the internal list is scrolled and an animation indicating that we already are at the top of the list is showed.
 - There is an arrow in the middle of the pipeline that should show an icon of the action and of the list of reactions, but icons are not displayed.

**Review by AnonymusRaccoon — APPROVED** [review]

File: mobile/I10n.yaml (hunk 4 above)
Comment (Arthi-chaud) [reply mobile/I10n.yaml:3]: Will be fixed in the Translation-module PR

**Pull request merged**